server/datastore/mysql/migrations/tables/20230202224725_CreateHostDiskEncryptionKeysTable.go

```diff
@@ -0,0 +1,33 @@
+package tables
+
+import (
+	"database/sql"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230202224725, Down_20230202224725)
+}
+
+func Up_20230202224725(tx *sql.Tx) error {
+	// `decryptable` can be NULL to signal that we have fetched the key but
+	// we don't know yet if we can decrypt it or not, the related index is
+	// to aid querying for this scenario by taking adventage of MySQL's IS
+	// NULL optimization:
+	// https://dev.mysql.com/doc/refman/5.7/en/is-null-optimization.html
+	_, err := tx.Exec(`
+	  CREATE TABLE IF NOT EXISTS host_disk_encryption_keys (
+            host_id             int(10) UNSIGNED NOT NULL,
+	    base64_encrypted    blob NOT NULL,
```

Comment:
Since it's base64, shouldn't we define it as text/varchar?

Reply:
I thought about that, reading through https://dev.mysql.com/doc/refman/8.0/en/blob.html and this bit jumped at me:

> BLOB values are treated as binary strings (byte strings). They have the binary character set and collation, and comparison and sorting are based on the numeric values of the bytes in column values. TEXT values are treated as nonbinary strings (character strings). They have a character set other than binary, and values are sorted and compared based on the collation of the character set.

so `BLOB` make a tiny bit more sense, as it's just binary encoded data without a real text meaning, but I think either will do a fine job, I don't mind changing it!

Reply:
yeah that was really a nit, just to avoid blob if we could but for no other reason than it might have some trickiness or non-obvious behavior to work with/decode into structs, etc. But I certainly don't think the data type itself should be a problem for storing the data. No problem to leave like it is.

Reply:
I'll trust your instinct here, updated!